packages/patterns/src/types.js

```diff
@@ -480,10 +480,16 @@ export {};
  * @typedef {{
  * <M extends Record<any, MethodGuard>>(interfaceName: string,
  *             methodGuards: M,
- *             options?: {sloppy?: false}): InterfaceGuard<M>;
+ *             options?: {
+ *     sloppy?: false,
```

Comment:
@turadg , since I'm unsure why you split these cases on `sloppy` true vs false, I may be doing the wrong thing in not splitting on unprotected true vs false. Truly there are three cases: (false,false), (true,false), and (false,true). If I had not already introduced a boolean option, I would have made it a three-way. Oh well.

Reply:
The reason for the overloading here is that `sloppy:true` implies `InterfaceGuard<any>` (everything matches) and `sloppy:false` parameterizes InterfaceGuard with the type of `methodGuards`.

I take it that "unprotected" should be `InterfaceGuard<any>` so this type needs more work. But let's settle on the API first.

> I would have made it a three-way

It's not too late! We could make it `sloppiness: `none' | 'sloppy' | 'unprotected'` and still retain backwards compatibility with `true` ('sloppy') and `false` ('none')